### DISCUSSION TITLE
Synchronize a typst document with an external data file

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#3040] [Q&A] [answered]
Consider the following code:

```
#let results = csv("data.csv")

#table(
  columns: 2,
  [*Condition*], [*Result*],
  ..results.flatten(),
)
```

The file `data.csv` is being generated by a `python` script, external to the `typ` document. So, my question is: How to manage to avoid non-existence errors while the file is not yet created by the `python` script. Moreover, how to do so that the reading of the `csv` file is done as soon as it changes?

**Comment by Enivex** [comment, marked as answer]
> How to manage to avoid non-existence errors while the file is not yet created by the python script. 

Just create an empty file

> Moreover, how to do so that the reading of the csv file is done as soon as it changes?

Typst does that for you automatically if you use typst watch

**Comment by psads-git** [comment]
Thanks a lot, @Enivex: That works. I was speculating that some state variable would be necessary, but it is much easier!
